### DISCUSSION TITLE
No output generated in ephemeral container on Kubernetes

### DISCUSSION THREAD

**Discussion by Danny02** [async-profiler/async-profiler#904] [Q&A] [answered]
Hi,

I have a Java application running on Kubernetes that is based on the `eclipse-temurin:21-jdk`image. In my thinking I should be able to start a profiling session in a debug container (based on `ubuntu:20.04` with `async-profiler-3.0-linux-x64.tar.gz`).

So I execute:
`kubectl debug -it my-pod --image=profiling-tools --target=application`

```
root@my-pod:/usr/local/bin# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1 51.8 11.6 17839688 3764692 ?    Ssl  Mar12 542:06 java --enable-preview -cp @/app/jib-classpath-file com.my.App
root      226703  0.0  0.0   5036  4320 pts/0    Ss   08:16   0:00 /bin/bash
root      227271  0.0  0.0   5908  2912 pts/0    R+   08:17   0:00 ps aux
```

```
root@my-pod:/usr/local/bin# jps
1 App
228444 Jps
```

```
root@my-pod:/usr/local/bin# asprof -d 5 -f /tmp/profile.html 1
Profiling for 5 seconds
Done
```

No `/tmp/profile.html` file is created (even I can write to `/tmp`). I do not get any other console output without `-f` or with `-L debug`.

Does anyone have any idea? Missing libraries, wrong configuration, Pod security settings ....

**Comment by apangin** [comment, marked as answer]
`-f /tmp/profile.html` specifies the path inside the target container, not in the namespace where you run `asprof`.
Check if you can access `/proc/1/root/tmp/profiler.tmp`
